### PR TITLE
Add getRangeFontName to TextNodeStub

### PR DESCRIPTION
**Pull request by abhijeetnarvekar**
Hi,

I have proposed a change to add `getRangeFontName` to `TextNodeStub`
I have just returned the existing font information or the default.

Let me know if your thoughts.

Thanks.

### CODE REVIEW

File: src/stubs.ts [hunk 1]
```diff
@@ -324,6 +328,15 @@ export const createFigma = (config: TConfig): PluginAPI => {
...
       this._textAutoResize = value;
     }
+    getRangeFontName(
+      start: number,
+      end: number
+    ): FontName | PluginAPI["mixed"] {
+      if (start && end && end >= start) {
+        return this._fontName || { family: "Roboto", style: "Regular" };
+      }
+      throw new Error(`Error: Invalid range: ${start}, ${end}`);
```
Comment (ilyalesik) [inline src/stubs.ts]: `if (joinedConfig.simulateErrors)` is needed here

File: src/stubs.ts [hunk 2]
```diff
@@ -233,6 +233,10 @@ export const createFigma = (config: TConfig): PluginAPI => {
   }
 
   class LayoutMixinStub implements LayoutMixin {
+    layoutGrow: number;
+    rescale(scale: number): void {
+      return;
```
Comment (ilyalesik) [inline src/stubs.ts]: Maybe we also need to do scaling:
```ts
this.width = this.width * scale;
this.height = this.height * scale;
```

**Review by ilyalesik — CHANGES_REQUESTED** [review]
The project uses `yarn.lock` instead of `package-lock.json`

**Comment by ilyalesik** [comment]
Hey, @abhijeetnarvekar! Thank you very much for the PR. Could you make changes by comments, please?

File: src/stubs.ts (hunk 1 above)
Comment (abhijeetnarvekar) [reply src/stubs.ts]: Added 'joinedConfig.simulateErrors' with error handling for invalid start, end values.

File: src/stubs.ts (hunk 2 above)
Comment (abhijeetnarvekar) [reply src/stubs.ts]: Applied your suggestion along with error handling for rescale factor to be >= 0.01

**Comment by abhijeetnarvekar** [comment]
Thanks for your review. I have made the suggested changes and removed package.lock.json. Let me know your feedback.

**Review by ilyalesik — APPROVED** [review]

**Comment by ilyalesik** [comment]
Thanks! Published `0.0.46`

**Pull request merged**